sui_types/src/object.rs

```diff
@@ -400,6 +400,20 @@ impl Object {
         ObjectDigest::new(sha3_hash(self))
     }
 
+    // Size of the object in bytes.
+    // TODO: For now we just look at the size of the data.
+    // Do we need to be accurate and look at the serialized size?
+    pub fn object_data_size(&self) -> usize {
+        match &self.data {
```

Comment:
Should we include the metadata size as well?

Reply:
That's where it becomes tricky. To chase down the size of the metadata, we need to recursively go into each metadata struct and manually calculate their sizes. For instance, for Move object, we have a type_tag, which then contains Identifier of the module and function, and each Identifier is a Box of str and etc.

Reply:
Ah, got it + good point... 

I think we can implement a `StructTag::length()` function (or similar) that tells you how many characters are in the module name and function name. And it's probably worthwhile to do so because I can imagine many common cases where the length of the struct name and metadata dwarfs the size of the actual data (e.g., `Coin<T>` will behave like this). Not a blocker for this PR, of course.